cmd/createorupdate/createorupdate.go

```diff
@@ -13,47 +16,147 @@ import (
 	"github.com/openshift/openshift-azure/pkg/fakerp"
 )
 
-func main() {
-	logger := logrus.New()
-	logger.Formatter = &logrus.TextFormatter{FullTimestamp: true}
-	logger.SetLevel(logrus.DebugLevel)
-	log := logrus.NewEntry(logger)
+type server struct {
+	// the server will not process more than a single
+	// request at all times.
+	inProgress chan struct{}
+	log        *logrus.Entry
+}
+
+// ServeHTTP handles an incoming request to the server.
+func (s *server) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	defer req.Body.Close()
 
-	// read in the external API manifest.
-	b, err := ioutil.ReadFile("_data/manifest.yaml")
+	ok := s.validate(w, req, nil)
+	if !ok {
+		return
+	}
+	defer func() {
+		// drain once we are done processing this request
+		<-s.inProgress
```

Comment:
How does this work?  It is sending on the inProgress chan struct but I'm not sure I understand how it works.
```
inProgress: make(chan struct{}, 1)}
```

Reply:
This is a receive from the channel to drain it in order to start accepting new requests.